misc/python/materialize/mzcompose/composition.py

```diff
@@ -329,25 +331,38 @@ def invoke(
                         close_fds=False,
                         stdin=subprocess.PIPE,
                         stdout=subprocess.PIPE,
-                        stderr=subprocess.STDOUT,
+                        stderr=subprocess.PIPE,
                         text=True,
                         bufsize=1,
                     )
                     if stdin is not None:
                         p.stdin.write(stdin)  # type: ignore
                     if p.stdin is not None:
                         p.stdin.close()
-                    for line in p.stdout:  # type: ignore
-                        print(line, end="")
-                        stdout_result += line
+                    sel = selectors.DefaultSelector()
+                    sel.register(p.stdout, selectors.EVENT_READ)  # type: ignore
+                    sel.register(p.stderr, selectors.EVENT_READ)  # type: ignore
+                    while True:
+                        for key, val in sel.select():
+                            line = key.fileobj.readline()  # type: ignore
+                            if not line:
+                                break
+                            if key.fileobj is p.stdout:
+                                print(line, end="")
+                                stdout_result += line
+                            else:
+                                print(line, end="", file=sys.stderr)
+                                stderr_result += line
```

Comment:
You might need to use these vars in the `except` block as well, don't you?

Reply:
Oh, that catch block below actually explains the weirdness I have been seeing with some output coming way below other. Done.